File: ljh/java/src/level1/ReverseNumber.java
package level1;

/*
 * 자연수 뒤집어 배열로 만들기 (220108 - 220114) 
 */

public class ReverseNumber {
	
	public int[] solution(long n) {
		int[] answer = new int[(int)(Math.log10(n)+1)];
		
		for(int i=0; i<answer.length; i++) {
			answer[i] = (int)(n%10);
			n /= 10;
		}
		
		return answer;
	}
}

File: ljh/javascript/level1/ReverseNumber.js
// 자연수 뒤집어 배열로 만들기 (220108 - 220114)
// solution2 > solution3 > solution1
export default class ReverseNumber {
    solution1 = (n) => {
        const arr = [];
        [...n+""].forEach(v =>arr.unshift(+v));
        return arr;
    }

    solution2 = (n) => {
        const arr = [];
        while(n > 0){
            arr.push(n % 10);
            n = Math.floor(n/10);
        }
        return arr;
    }

    solution3 = (n) => {
        const arr = [];
        const numberArr = [...n+""];
        for(let i=numberArr.length-1; i>=0; i--) arr.push(+numberArr[i]); 
        return arr;
    }
}